app/views/events/index.html.erb

```diff
@@ -0,0 +1,23 @@
+<h1>Events</h1>
+
+<h3>FUTURE EVENTS</h3>
+<% Event.future.each do |event| %>
```

Comment:
- lists past events, i.e. @past_events = Event.previous_events ([screenshot from Odin](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png))
- lists upcoming events, i.e. @upcoming_events = Event.upcoming_events ([screenshot from Odin ](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png))

Rather than do `Event.future.each` you can do the above in your Event controller show action and later in your view just call `@upcoming_events.each`

Do this for other parts of your code with similar patterns. I can see you repeated such also in your Users show view. Check `users/profile.html.erb`